### PR TITLE
changed web3 on vulnerabilities page

### PR DESCRIPTION
**Pull request by JannikSommer**

### CODE REVIEW

File: frontend/src/components/vulnerabilities/Vulnerabilities.js
Comment (mlund98) [inline frontend/src/components/vulnerabilities/Vulnerabilities.js:1]: There is a lot of code in this file. Would it be possible to move some of the functionality to a seperate file? For instance, all the callback functions together with the vulnerability hook could be separated.

File: frontend/src/components/confidentialAnnouncement/ConfidentialAdvisoryForm.js [hunk 1]
```diff
@@ -119,8 +119,7 @@ function ConfidentialAdvisoryForm({accounts, ipfs}) {
                 default:
                     throw new Error("No storage system selected.");
             }
-
-            const result = await web3Gateway.announcePrivateSecurityAdvisory(
+            const result = await new Web3Gateway().announcePrivateSecurityAdvisory(
```
[the remark this answers is not in the document]
Comment (JannikSommer) [reply frontend/src/components/confidentialAnnouncement/ConfidentialAdvisoryForm.js:122]: Both implementations are okay. The one in `app.js` is used to keep already discovered events, such that we do not need to retrieve all events in the blockchain.

File: frontend/src/components/announcement/UpdateAdvisoryForm.js [hunk 2]
```diff
@@ -87,7 +87,7 @@ function UpdateAdvisoryForm({ accounts, ipfs }) {
             }
             await parseCSAF();
 
-            const result = await web3Gateway.announcePublicSecurityAdvisoryUpdate(
+            const result = await new Web3Gateway().announcePublicSecurityAdvisoryUpdate(
```
[the remark this answers is not in the document]
Comment (JannikSommer) [reply frontend/src/components/announcement/UpdateAdvisoryForm.js:90]: Both implementations are okay. The one in `app.js` is used to keep already discovered events, such that we do not need to retrieve all events in the blockchain.

File: frontend/src/components/announcement/NewAdvisoryForm.js [hunk 3]
```diff
@@ -90,8 +90,7 @@ function NewAdvisoryForm({ accounts, ipfs }) {
                     throw new Error("No storage system selected.");
             }
             await parseCSAF();
-
-            let res = await web3Gateway.announcePublicSecurityAdvisory(
+            let res = await new Web3Gateway().announcePublicSecurityAdvisory(
```
[the remark this answers is not in the document]
Comment (JannikSommer) [reply frontend/src/components/announcement/NewAdvisoryForm.js:93]: Both implementations are okay. The one in `app.js` is used to keep already discovered events, such that we do not need to retrieve all events in the blockchain.

File: frontend/src/models/web3/web3Gateway.js [hunk 4]
```diff
@@ -2,118 +2,86 @@ import Web3 from "web3";
...
+        this.subscriptions.push(this.announcementService.events.NewSecurityAdvisory(
+            {fromBlock: 0}, 
+            callback)
+        );
     }
 
     /**
      * Subscribe to updated security advisory events on the Announcement Service.
-     * @param {*} advisoryIdentifier Identifier from the NewSecurityAdvisory event to subscribe to updates for.
+     * @param {string} advisoryIdentifier Identifier from the NewSecurityAdvisory event to subscribe to updates for.
```
[the remark this answers is not in the document]
Comment (JannikSommer) [reply frontend/src/models/web3/web3Gateway.js:66]: As of [this](https://jsdoc.app/tags-param.html) it should be lower case.

File: frontend/src/models/web3/web3Gateway.js [hunk 5]
```diff
@@ -2,118 +2,86 @@ import Web3 from "web3";
...
-                cid: event.returnValues.documentLocation,
-                tx: await this.web3.eth.getTransaction(event.transactionHash), 
-                block: await this.web3eth.getBlock(event.blockNumber),
-                advisory: {} // will be set later
-            };
-            this.newSecurityAdvisoryEvents.push(newEvent);
-            await this.subscribeToSecurityAdvisoryUpdates(event.returnValues.advisoryIdentifier);
-        });
-        this.subscriptions.push(sub);
+     * @param {function} Callback function for when events are found.
```
[the remark this answers is not in the document]
Comment (JannikSommer) [reply frontend/src/models/web3/web3Gateway.js:55]: As of [this](https://jsdoc.app/tags-param.html) it is written wrong.

File: frontend/src/models/web3/web3Gateway.js
[the remark this answers is not in the document]
Comment (JannikSommer) [reply frontend/src/models/web3/web3Gateway.js:1]: Not all should start with capital letter.

**Pull request merged**